lib/api/apiUtils/object/versioning.js

```diff
@@ -129,10 +166,14 @@ export function versioningPreprocessing(bucketName, bucketMD, objectKey,
             if (nullVersionId === undefined) {
                 return callback(null, options);
             }
-            options.deleteNullVersionData = true;
             return _deleteNullVersionMD(bucketName, objectKey,
-                { versionId: nullVersionId }, log,
-                err => callback(err, options));
+                { versionId: nullVersionId }, log, (err, nullDataToDelete) => {
+                    if (err === 'NoSuchVersion') {
+                        return callback(errors.InternalError);
```

Comment:
before if metadata unexpectedly could not find the null version to update MD / delete data, we were returning NoSuchKey error. (If we keep the check, should check for 'NoSuchKey', not 'NoSuchVersion') This meant if user is trying to put a version, complete a mpu or do object copy and we get this unexpected error, they would get 'NoSuchKey', which might be confusing to the user.

The alternative is we could silence the error and put a new version anyway, which may succeed, but the previous null version may unexpectedly show up in listings if deleting the metadata failed.